File: projects/GPL-FH-Java/features/DirectedOnlyVertices/GPL/Vertex.java
package GPL;

// dja - trying to fix compile problems
import java.util.Iterator;
import java.util.LinkedList;

// ************************************************************
   
public class Vertex implements EdgeIfc, NeighborIfc
{
    public LinkedList adjacentVertices;
    public String name;
 
    public Vertex() {
        VertexConstructor();
    }
  
    public void VertexConstructor() {
        name      = null;
        adjacentVertices = new LinkedList();
    }

    public  Vertex assignName( String name ) {
        this.name = name;
        return ( Vertex ) this;
    }

    //dja: fix for compile errors during performance improvements
    public String getName( ) 
    { 
        return name; 
    }

 
    public void addAdjacent( Vertex n ) {
        adjacentVertices.add( n );
    }

    public void adjustAdorns( Vertex the_vertex, int index ) 
      {}
      
    // dja - trying to fix compile errors
    public VertexIter getNeighbors( ) 
    {
        return new VertexIter( ) 
        {
            private Iterator iter = adjacentVertices.iterator( );
            public Vertex next( ) 
            { 
               return ( Vertex )iter.next( ); 
            }

            public boolean hasNext( ) 
            {
               return iter.hasNext( ); 
            }
        };
    }


    public void display() {
        int s = adjacentVertices.size();
        int i;

        System.out.print( "Vertex " + name + " connected to: " );

        for ( i=0; i<s; i++ )
            System.out.print( ( ( Vertex )adjacentVertices.get( i ) ).name+", " );
        System.out.println();
    }

//--------------------
// from EdgeIfc
//--------------------

    public Vertex getStart( ) { return null; }
    public Vertex getEnd( ) { return null; }

    public void setWeight( int weight ){}
    public int getWeight() { return 0; }

    public Vertex getOtherVertex( Vertex vertex )
    {
        return this;
    }



    public void adjustAdorns( EdgeIfc the_edge )
    {
    }

}

File: plugins/de.ovgu.featureide.examples/featureide_examples/GPL-FH-CSharp/features/DirectedOnlyVertices/Vertex.cs
//layer Directed;

// dja - trying to fix compile problems
//import java.util.Iterator;
using System.Collections;
using System.Collections.Generic;

// ************************************************************
   
public class Vertex : EdgeIfc, NeighborIfc
{
    public List<Vertex> adjacentVertices;
    public string name;
 
    public Vertex() {
        VertexConstructor();
    }
  
    public void VertexConstructor() {
        name      = null;
        adjacentVertices = new List<Vertex>();
    }

    public  Vertex assignName( string name ) {
        this.name = name;
        return this;
    }

    //dja: fix for compile errors during performance improvements
    public string GetName( ) 
    { 
        return name; 
    }

 
    public void AddAdjacent( Vertex n ) {
        adjacentVertices.Add( n );
    }

    public void adjustAdorns( Vertex the_vertex, int index ) 
      {}
      
    // dja - trying to fix compile errors
    public Iterator<Vertex> GetNeighbors( ) {
    	return new Iterator<Vertex>(adjacentVertices.GetEnumerator());
    }
    public void display() {
        int s = adjacentVertices.Count;
        int i;

        System.Console.Out.Write( "Vertex " + name + " connected to: " );

        for ( i=0; i<s; i++ )
            System.Console.Out.Write( (adjacentVertices[i] ).name+", " );
        System.Console.Out.WriteLine();
    }

//--------------------
// from EdgeIfc
//--------------------

    public Vertex GetStart( ) { return null; }
    public Vertex GetEnd( ) { return null; }

    public void setWeight( int weight ){}
    public int GetWeight() { return 0; }

    public Vertex GetOtherVertex( Vertex vertex )
    {
        return this;
    }



    public void adjustAdorns( EdgeIfc the_edge )
    {
    }

}
